analysis.tex

```diff
@@ -1,29 +1,53 @@
 
-\begin{theorem}
-  If $H$ is modelled as a \emph{Random Oracle}, then the protocol $\Pi$ of Section~\ref{section:construction} is \emph{unspendable}.
+\begin{theorem}[Correctness]
+  The Proof-of-Burn protocol $\Pi$ of Section~\ref{section:construction} is \emph{correct}.
 \end{theorem}
 \begin{proof}
-  $H(t) \xor 1 = pkh$.
-  Adversary knows a $pk$ st $H(pk) = pkh$.
-  This is impossible under the Random Oracle model.
-  TODO
+  We need to prove that $\forall \kappa, t. \BurnVerify(1^\kappa, t, \GenBurnAddress(1^\kappa, t)) = \textsf{true}$.
+
+  Based on Algorithm~\ref{alg.construction}, $\BurnVerify(1^\kappa, t, \GenBurnAddress(1^\kappa, t)) = \textsf{true}$ if and only if $\GenBurnAddress(1^\kappa, t) = \GenBurnAddress(1^\kappa, t)$, which always holds as $\GenBurnAddress$ is deterministic.
+\end{proof}
+
+\begin{theorem}[Unspendability]
+  If $H$ is modelled as a preimage-resistant hash function, then the protocol $\Pi$ of Section~\ref{section:construction} is \emph{unspendable}.
+\end{theorem}
+\begin{proof}
+  Assume that the protocol $\Pi$ is not unspendable. This means that there exists a probabilistic polynomial-time adversary $\mathcal{A}$ such that:
+
+  \[
+    \Pr[\textsc{spend-attack}_{\mathcal{A},
+    \Pi}(\kappa) = \textsf{true}] > negl(\kappa)
+  \]
+
+  This would imply that the adversary can generate $t$, $m$, $\sigma = (pk, \sigma')$ and $pkh$ such that $\BurnVerify(1^\kappa, t, pkh) = \true \land \SpendVerify(m, \sigma, pkh) = \true$ with non-negligible probability in $\kappa$.
+
+  From the construction we know that $pkh = H_\kappa(t) \xor 1$.
+
+  For $\SpendVerify(m, \sigma, pkh) = \true$, it should be so that $H_\kappa(pk) = pkh = H_\kappa(t) \xor 1$.
+
+  From the preimage resistance of $H_\kappa$ we know that given a $pkh$ a probabilistic polynomial-time adversary should be able to discover an input $pk$ with negligible probability in $\kappa$.
 \end{proof}
 
 We note that the security of the signature scheme is not needed to prove unspendability. Were the signature scheme of Bitcoin ever found to be forgeable, the coins burned through our scheme would remain unspendable.
 
-\begin{theorem}
+\import{./}{algorithms/alg.collision-resistance-adversary.tex}
+
+\begin{theorem}[Binding]
   If $H$ is \emph{collision resistant} then the protocol of Section~\ref{section:construction} is \emph{binding}.
 \end{theorem}
 \begin{proof}
   Let $\mathcal{A}$ be an arbitrary adversary against $\Pi$.
-  We will construct the Collision Resistance adversary $A^*$ against $H$.
-  TODO
+  We will construct the Collision Resistance adversary $\mathcal{A}^*$ against $H$.
+
+  The collision resistance adversary, illustrated in Algorithm~\ref{alg.collision-resistance-adversary}, calls $\mathcal{A}$ and obtains two outputs, $t$ and $t'$ such that $H(t) \xor 1 = H(t') \xor 1 \implies H(t) = H(t')$.
+
+  If $\mathcal{A}$ is successful in the \textsf{Bind} game then $\mathcal{A^*}$ is successful in the \textsf{Collision} game.
```

Comment:
What you state here implies a `$\leq$` in the probability relation. If you want `$=$` then you must say iff.